src/NHSD.BuyingCatalogue.Identity.Api/Services/LoginService.cs

```diff
@@ -57,13 +90,32 @@ private async Task RaiseLoginSuccessAsync(string username, AuthorizationRequest
             await _eventService.RaiseAsync(new UserLoginSuccessEvent(user.UserName, user.Id, user.UserName, clientId: context?.ClientId));
         }
 
-        private async Task<bool> SignInAsync(string username, string password, AuthorizationRequest context)
+        private async Task<bool> ValidateUserAsync(ApplicationUser user, string password, AuthorizationRequest context)
         {
-            var result = await _signInManager.PasswordSignInAsync(username, password, false, true);
+          var result = await _signInManager.CheckPasswordSignInAsync(user, password, true);
             if (result.Succeeded)
+            {
+                var isDisabled = await ValidateIfUserDisabledAsync(user, context);
+
+                return isDisabled;
+            }
+            
+            await _eventService.RaiseAsync(new UserLoginFailureEvent(user.UserName, ValidateUserMessage, clientId: context?.ClientId));
+            errors.Add(LoginUserErrors.UserNameOrPasswordIncorrect());
+
+            return false;
+        }
+
+        private async Task<bool> ValidateIfUserDisabledAsync(ApplicationUser user, AuthorizationRequest context)
```

Comment:
Thinking of the single responsibility pattern here. The validate method is checking whether or not a user is disabled, but it is also raising an event that is associated with the sign in process. Do you think these two responsibilities could be split out?